system-design/app-interoperability/application-package-definition.md

```diff
@@ -17,286 +33,188 @@ The application package has the following folder structure:
+| email            | string          | N               | Email address of the application's creator.|
 
-Below, the usage of Helm as an application manifest of an Margo application is defined. This can be chosen if the target edge device is running Kubernetes behind the workload orchestration agent. For a full documentation of Helm, we refer to its official [website](https://helm.sh/docs/).
+**Organization Metadata Attributes**
 
-A Helm chart includes the following Helm-specific configuration files and Kubernetes manifest templates:
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| name             | string          | Y               | Organization responsible for the application's development and distribution.|
+| site             | string          | N               | Link to the organization's website.|
 
-- Chart.yaml
-This file contains metadata about the chart such as name, version, description, and maintainers.
-- values.yaml
-This file defines the configuration values for the templates. Using this template approach, only this file needs to be changed for deployments in different environments (e.g., development, staging or production).
-- templates
-This directory contains all Kubernetes manifest files templated to be instantiated with values from the values.yaml file:
-  - deployment.yaml
-  - service.yaml
-  - configmap.yaml
-  - Secrets
-  - and other Kubernetes resources.
-- charts
-This directory may include additional charts that are necessary for the parent chart to function.
+**Deployment Attributes**
 
-> **Note**  
-> This version of the Margo specification uses Helm version 3. For additional information refer to the Helm Documentation at [https://helm.sh](https://helm.sh).
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| type             | string          | Y               | Indicates the components's deployment configuration. The values are `helm.v3` to indicate the component's package format is Helm version 3 and `docker-compose` to indicate the component's package format is Docker Compose. When installing the application on a device supporting the Kubernetes platform all `helm.v3` components, and only `helm.v3` components, will be provided to the device in same order they are listed in the application description file. When installing the application on a device supporting docker-compose all `docker-compose` components, and only `docker-compose` components, will be provided to the device in the same order they are listed in the application description file. The device will install the components in the same order they are listed in the application description file. Component types under `cluster` must use `helm.v3`. Component types under `standalone` must use `docker-compose` |
```

Comment:
Do we consider supporting mixture of component types in the future, such as a Helm chart describing the backend, and an Android app representing the frontend?

Reply:
@Haishi2016 Good question. I think this is something we'll have to see over time. I don't think we need to address it right now since we're still in a pre-draft state. If someone brings a use case for it that we need to support then we can explore it.

@arne-broering maybe this is something we could put in the roadmap just to explore it as an option to see if it's needed, or not, and how it should be supported if it is needed.

Reply:
yes, sounds good. I'll add it to the project's roadmap.